### ISSUE TITLE
wait for informer caches to sync before processing events

### ISSUE THREAD

**Issue by skriss** [projectcontour/contour#3020]
Contour does not consistently wait for its informer caches to sync before processing events coming from them. Doing this is a best practice in Kubernetes controller development and can help avoid issues related to those caches not being fully populated (e.g. items missing from the listers). We should look into all the event handlers within Contour, and ensure that they're appropriately waiting for the caches to sync before starting to process events.

**Comment by jpeach** [comment]
Probably either a subtask or duplicate of https://github.com/projectcontour/contour/issues/1280

**Comment by skriss** [comment]
This is obsolete given the recent migration to controller-runtime, closing.
